include/drivers/sensor/sht4x.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2021, Leonard Pollak
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Extended public API for Sensirion's SHT4X T/RH sensors
+ *
+ * This exposes an API to the on-chip heater which is specific to
+ * the application/environment and cannot be expressed within the
+ * sensor driver abstraction.
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_SENSOR_SHT4X_H_
+#define ZEPHYR_INCLUDE_DRIVERS_SENSOR_SHT4X_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <drivers/sensor.h>
+
+/* Maximum teperature above which the heater should not be used */
+#define SHT4X_HEATER_MAX_TEMP 65
+
+enum sht4x_heater_power {
+	/* High power heater pulse -> ~200 mW  @3.3V */
+	SHT4X_HEATER_POWER_HIGH,
+	/* Medium power heater pulse -> ~110 mW  @3.3V */
+	SHT4X_HEATER_POWER_MED,
+	/* Low power heater pulse -> ~20 mW  @3.3V */
+	SHT4X_HEATER_POWER_LOW
+};
+
+enum sht4x_heater_duration {
+	/* Long heater pulse -> 1.1s */
+	SHT4X_HEATER_DUR_LONG,
+	/* Short heater pulse -> 0.11s */
+	SHT4X_HEATER_DUR_SHORT
+};
+
+
+/**
+ * @brief Fetches data using the on-chip heater.
+ *
+ * Measurement is always done with "high" repeatability.
+ *
+ * @param dev Pointer to the sensor device
+ * @param power the heater power
+ * @param duration the length of the heating pulse
+ *
+ * @return 0 if successful, negative errno code if failure.
+ */
+int sht4x_fetch_with_heater(const struct device *dev,
+		enum sht4x_heater_power power,
+		enum sht4x_heater_duration duration);
```

Comment:
This can be solved using the sensor attributes API. One idea:

1. an attribute to set power
2. an attribute to set duration
3. When fetching, if previous power/duration setting was changed, configure it first.

I think custom APIs should be introduced when there is really no other solution.

Reply:
I am fine with getting rid of the public APIs.
for the SGP40 it will be be easy (setting T/RH via attributes API).
however in this case i am not quite sure on how to activate the heater while getting rid of this function.